lightning/src/routing/router.rs

```diff
@@ -876,40 +878,126 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
+							aggregate_next_hops_fee_msat = match aggregate_next_hops_fee_msat.checked_add(final_value_msat) {
+								Some(value) => value,
+								None => u64::max_value()
+							};
+						} else {
+							hop_used = false;
+						};
+					} else {
+						// If this is not the last hop then check if it is a feasible intermediate hop
+						// otherwise test the previous hop for direct channel path and break.
+						let reqd_channel_cap = Some(match aggregate_next_hops_fee_msat.checked_add(999) {
+							Some(cap) => cap,
+							None => u64::max_value()
+						} / 1000); // converting from msat
+
+						if !add_entry!(hop.short_channel_id, hop.src_node_id, prev_hop, directional_info, reqd_channel_cap, &empty_channel_features, aggregate_next_hops_fee_msat, path_value_msat, aggregate_next_hops_path_htlc_minimum_msat) {
+							hop_used = false;
+						}
+					}
+
+					if !hop_used {
+						// If this hop was not used then there is no use checking the preceeding hops
+						// in the RouteHint. We can break by just searching for a direct channel between 
+						// last checked hop and first_hop_targets
+						if let Some(&(ref first_hop, ref features, ref outbound_capacity_msat, _)) = first_hop_targets.get(&prev_hop) {
+							add_entry!(first_hop, *our_node_id , prev_hop, dummy_directional_info, Some(outbound_capacity_msat / 1000), features, aggregate_next_hops_fee_msat, path_value_msat, aggregate_next_hops_path_htlc_minimum_msat);
+						}
+						break;
+					}
+
+					if idx == route.len() - 1 {
+						// The last hop in this iterator is the first hop in
+						// overall RouteHint.
+						// If this hop connects to a node with which we have a direct channel,
+						// ignore the network graph and, if the last hop was added, add our
+						// direct channel to the candidate set.
+						//
+						// Note that we *must* check if the last hop was added as `add_entry`
+						// always assumes that the third argument is a node to which we have a
+						// path.\
+						if let Some(&(ref first_hop, ref features, ref outbound_capacity_msat, _)) = first_hop_targets.get(&hop.src_node_id) {
+							add_entry!(first_hop, *our_node_id , hop.src_node_id, dummy_directional_info, Some(outbound_capacity_msat / 1000), features, aggregate_next_hops_fee_msat, path_value_msat, aggregate_next_hops_path_htlc_minimum_msat);
+						}
 					}
+
+					aggregate_next_hops_fee_msat = match aggregate_next_hops_fee_msat.checked_add(match idx {
+
+						// When its the Last Hop in the RouteHint, we want the aggregate fees to
+						// be the sum of the final value to route and the fees to route through
+						// the channel between the Last Hop Node and the Payee Node.
+						0 => match compute_fees(aggregate_next_hops_fee_msat, hop.fees) {
```

Comment:
This and the `_` block are identical?

Reply:
Yes, for case 0 I was adding the final_value_msat here but then switched to do it inside the add_entry's if block for idx == 0. New commit reflects this change.